tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -24,6 +24,9 @@ sealed trait NativeConfig {
   /** The compilation options passed to LLVM. */
   def compileOptions: Seq[String]
 
+  /** Target triple that defines current OS, ABI and CPU architecture. */
+  def targetTriple: String
```

Comment:
Personally, I would go with `Option[String]` and `withTargetTriple(value: String)`. All `nativeX` settings including `nativeTargetTriple` exist only for backward compatibility and would be removed in the future. The only remaining way to interact with them would be done via `nativeConfig` in the same way as it's done in scala.js.